src/cmd-generate-hashlist

```diff
@@ -0,0 +1,179 @@
+        :type timestamp: str
+        :raises: KeyError
+        """
+        super()
+        self._metadata = metadata
+        if release is None:
+            release = self._metadata['buildid']
+        self['release'] = release
+
+        self['meta'] = {}
+        if generator is None:
+            generator = 'coreos-assembler-' + self._metadata['coreos-assembler.container-image-git']['commit']  # noqa
+        self['meta']['generator'] = generator
+        if timestamp is None:
+            timestamp = datetime.datetime.utcnow().isoformat()
+        self['meta']['timestamp'] = timestamp
+        self['meta']['image'] = os.path.sep.join([
+            'builds', self._metadata['ostree-version'],
+            self._metadata['coreos-assembler.basearch'],
+            self._metadata['images']['qemu']['path']])
+
+        self['hashes'] = {}
+
+    def populate_hash_list(self):
+        """
+        Populates the hashlist instance with ostree and initramfs hashes.
+
+        :param checksum: The ostree checksum from the metadata
+        :type checksum: str
+        :param commit: The ostree commit
+        :type commit: str
+        :param commit_tar: Path to the ostree tar from the metadata
+        :type commit_tar: str
+        :param hashlist: Initialized hash list instance
+        :type hashlist: HashListV1
+        :returns: Nothing
+        :rtype: None
+        """
+        checkout = 'tmp/repo/tmp/keylime-checkout'
+        commit_tar = os.path.join(
+            self._metadata.build_dir,
+            self._metadata['images']['ostree']['path'])
+
+        import_ostree_commit(
+            'tmp/repo',
+            self._metadata['ostree-commit'],
+            commit_tar,
+            force=True)
+        subprocess.check_call([
+            'ostree', 'checkout',
+            '--repo=tmp/repo', '-U',
+            self._metadata['ostree-commit'], checkout])
+        self.hash_from_path(checkout)
+
+        # Extract initramfs contents
+        initramfs_path = os.path.realpath(
+            os.path.join(
+                '.', checkout, 'usr/lib/modules/*/initramfs.img'))
+        with tempfile.TemporaryDirectory() as tmpdir:
+            curdir = os.getcwd()
+            os.chdir(tmpdir)
+            subprocess.check_call(
+                f'/usr/lib/dracut/skipcpio {initramfs_path} | gunzip -c | cpio -idmv',  # noqa
+                shell=True)
+            self.hash_from_path(tmpdir)
+            os.chdir(curdir)
+
+    def hash_from_path(self, toppath):
+        """
+        Create hashes from files starting at a specific path.
+
+        .. note:: If a file can not be read it is skipped
+
+        :param toppath: The starting path to go through
+        :type toppath: str
+        :returns: Nothing
+        :rtype: None
+        """
+        for dirpath, _, filenames in os.walk(toppath):
+            for fname in filenames:
+                filepath = os.path.join(dirpath, fname)
+                # Skip broken symlinks
```

Comment:
Instead of doing `try/except` to deal with symlinks, let's just call `os.path.islink()` and `continue` if it is? That would also allow us to error out on permission/`ENOENT` errors we really shouldn't hit when traversing the checkout.